pysatMagVect/_core.py

```diff
@@ -960,331 +1389,345 @@ def closed_loop_edge_lengths_via_footpoint(glats, glons, alts, dates, direction,
+                apex_location_info(plus_x, plus_y, plus_z, dates,
+                                   ecef_input=True)
+
+        minus_apex_x, minus_apex_y, minus_apex_z = \
+                apex_location_info(minus_x, minus_y, minus_z, dates,
+                                ecef_input=True)
+
+    # take difference in apex locations
+    apex_edge_length = np.sqrt((plus_apex_x - minus_apex_x)**2 +
+                               (plus_apex_y - minus_apex_y)**2 +
+                               (plus_apex_z - minus_apex_z)**2)
+    if return_geodetic:
+        return apex_edge_length, plus_h - minus_h
+    else:
+        return apex_edge_length
+
+def scalars_for_mapping_ion_drifts(glats, glons, alts, dates,
+                                   max_steps=None, e_field_scaling_only=None,
+                                   edge_length=None, edge_steps=None,
+                                   **kwargs):
     """
     Calculates scalars for translating ion motions at position
     glat, glon, and alt, for date, to the footpoints of the field line
     as well as at the magnetic equator.
-    
+
     All inputs are assumed to be 1D arrays.
-    
+
+    Parameters
+    ----------
+    glats : list-like of floats (degrees)
+        Geodetic (WGS84) latitude
+    glons : list-like of floats (degrees)
+        Geodetic (WGS84) longitude
+    alts : list-like of floats (km)
+        Geodetic (WGS84) altitude, height above surface
+    dates : list-like of datetimes
+        Date and time for determination of scalars
+    e_field_scaling_only : Deprecated
+    max_steps : Deprecated
+    edge_length : Deprecated
+    edge_steps : Deprecated
+
+    Returns
+    -------
+    dict
+        array-like of scalars for translating ion drifts. Keys are,
+        'north_zonal_drifts_scalar', 'north_mer_drifts_scalar', and similarly
+        for southern locations. 'equator_mer_drifts_scalar' and
+        'equator_zonal_drifts_scalar' cover the mappings to the equator.
+
+
+    """
+
+    if e_field_scaling_only is not None:
+        raise DeprecationWarning('e_field_scaling_only no longer supported.')
+    if max_steps is not None:
+        raise DeprecationWarning('max_steps no longer supported.')
+    if edge_length is not None:
+        raise DeprecationWarning('edge_length no longer supported.')
+    if edge_steps is not None:
+        raise DeprecationWarning('edge_steps no longer supported.')
+
+    # use spacecraft location to get ECEF
+    ecef_xs, ecef_ys, ecef_zs = geodetic_to_ecef(glats, glons, alts)
+
+    # get footpoint location information
+    north_ftpnt, south_ftpnt = footpoint_location_info(ecef_xs, ecef_ys, ecef_zs,
+                                                       dates, ecef_input=True)
+
+    # prepare output memory
+    out = {}
+
+    # D and E vectors at user supplied location
+    # good for mapping to magnetic equator
+    _, _, _, _, _, _, _, _, _, infod = calculate_mag_drift_unit_vectors_ecef(ecef_xs, ecef_ys, ecef_zs, dates,
```

Comment:
This function is defined above and called like this every time.  Could we reconfigure the returned variables to aid in clarity?

Reply:
It is used elsewhere (step_along_..) with the unit vectors outputs assigned to variable names. More importantly, there is downstream code that depends upon pysatMagVect and changing the outputs here causes a ripple of work. Instead, I added a new method (calculate_geomagnetic_basis) with limited input options and an output dictionary that just transparently calls the calculate_mag_drift_unit_vectors. Existing code gets to remain the same and new code can use the simpler, clearer interface.